src/hook.rs

```diff
@@ -0,0 +1,133 @@
+use std::{slice, ptr, mem, str};
+use std::borrow::Cow;
+use libc::{self, c_int};
+use ffi::{self, lua_State, lua_Debug};
+use lua::extra_data;
+use util::callback_error;
+
+/// Contains information about the running code at the moments specified when setting the hook.
+/// Each field is described in the [Lua 5.3 documentaton][lua_doc].
+///
+/// Depending on when the hook is called, some fields might not be available. In these cases,
+/// integers and booleans might not be valid and/or strings are set to `None`.
+///
+/// [lua_doc]: https://www.lua.org/manual/5.3/manual.html#lua_Debug
+#[derive(Clone, Debug)]
+pub struct Debug<'a> {
+    pub name: Option<Cow<'a, str>>,
+    pub namewhat: Option<Cow<'a, str>>,
+    pub what: Option<Cow<'a, str>>,
+    pub source: Option<Cow<'a, str>>,
+    pub curr_line: u32,
+    pub line_defined: u32,
+    pub last_line_defined: u32,
+    pub num_ups: u32,
+    pub num_params: u32,
+    pub is_vararg: bool,
+    pub is_tailcall: bool,
+    pub short_src: Option<Cow<'a, str>>
+}
```

Comment:
Consider adding a private field here to allow rlua to add more fields without breaking users